`SideNav` - Extract a "base" component (#1321)
hashicorp/design-system: website/docs/components/side-nav/partials/code/component-api.md

<Doc::ComponentApi as |C|>
  <C.Property @name="<:header>" @type="named block">
    A named block where the content for the “header” area of the SideNav is rendered. The `SideNav::Header` component should be added here. It yields the value of `isMinimized` too.
  </C.Property>
  <C.Property @name="<:body>" @type="named block">
    A named block where the content for the “body” or main content of the SideNav is rendered. The `SideNav::List` and `SideNav::PortalTarget` components should be added here when used. It yields the value of `isMinimized` too.
  </C.Property>
  <C.Property @name="<:footer>" @type="named block">
    A named block where the content for the “footer” section of the SideNav is rendered. It yields the value of `isMinimized` too.
  </C.Property>
  <C.Property @name="...attributes">
    This component supports use of [`...attributes`](https://guides.emberjs.com/release/in-depth-topics/patterns-for-components/#toc_attribute-ordering).
  </C.Property>
</Doc::ComponentApi>

### SideNav::Base

This is the basic component (layout only).

<Doc::ComponentApi as |C|>
  <C.Property @name="<:root>" @type="named block">

>     A named block where content that lives in the "root" of the SideNav is rendered (the content should not cause a layout height).

Reviewer: "the content should not cause a layout height" could you please clarify this part? Does it mean the content shouldn't include a fixed height or something else?

Author: Yeah, you're right @KristinLBradley, it's not clear (it's hard to explain). Let me have a new go at tomorrow, with a fresh mind, and see how to improve the description.

Author: @KristinLBradley I've made a change that removes the issue at the "root" (pun intended) :)

>     A named block that can be used to render content that needs to be rendered outside of the "header/body/footer" containers of the SideNav.

Reviewer 2: suggestion – no strong opinion
```suggestion
    A named block for rendering content outside of the "header/body/footer" containers of the SideNav.
```
